mantidproject/mantid
# Re #20489 More reliable argument handling in StartLiveData

Python wrapper for StartLiveData now ensures that the `Connection` and `Listener` properties are set immediately after the `Instrument` property and before all other properties. 

This is to ensure that properties that are dependent on them, such as the dynamically loaded Listener Properties, are loaded correctly and not overwritten after being set.

**To test:**

Testing might be difficult, because the issue appears to be triggered by the order of the Python `dict` that contains the arguments passed to `StartLiveData`, which is undefined. But you can try the following, and hopefully it will fail the way it did on my machine.

Run the following in script interpreter (**must be connected to the STFC network**):
```python
StartLiveData(Instrument='ISIS_Kafka_Event',
                        Connection='kafka_event',
                        UpdateEvery=5,
                        PreserveEvents=True,
                        AccumulationMethod='Replace',
                        RunTransitionBehavior='Stop',
                        OutputWorkspace='ws',
                        InstrumentName='oldschema')
```

Before this change, this will usually fail with:

> Failed to find topic '_events' on broker

If you have the log set to debug, you will additionally see:

> Name: InstrumentName, Value: , Default?: Yes, Direction: Input

After the change, the listener will start and you will instead see, as expected: 

> Name: InstrumentName, Value: oldschema, Default?: No, Direction: Input

The started algorithm will run indefinitely, so you'll need to cancel it in the algorithm progress window.

Fixes #20489 

**Release Notes** 

Should this be in release notes? It's a bugfix for a small edge case issue that was found internally, not reported.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
